server/src/main/java/org/opensearch/search/ResourceType.java

```diff
@@ -8,12 +8,18 @@
 
 package org.opensearch.search;
 
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
 /**
  * Enum to hold the resource type
  */
+@PublicApi(since = "2.x")
```

Comment:
Shouldn't this also be experimental?

Reply:
SBP uses this construct already. So I think this should not be marked as Experimental